numpy/lib/function_base.py

```diff
@@ -1745,7 +1745,7 @@ def gradient(f, *varargs, **kwargs):
         slice4[axis] = slice(2, None)
```

Comment:
If up to me, I'd just create a base prefix `(slice(None),) * axis` here and add the 1,2,3,4 as need below - which I think would make for clearer code. But that is getting well beyond the immediate purpose of this PR, so feel free to ignore.